torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1837,6 +1858,20 @@ def forward(ctx, *deduped_flat_tensor_args):
                     assert isinstance(bw_state, BackwardState)
                     ctx._compiled_autograd_backward_state = bw_state
 
+                if num_rng:
+                    _curr_iter = next(curr_fwd_iter)
+                    ctx._curr_iter = _curr_iter
```

Comment:
a big comment explaining how we handle SAC rng with multiple fwd iterations before one bw iteration would be nice :)